code/app/src/main/java/com/example/bread/fragment/AddMoodEventFragment.java

```diff
@@ -261,4 +282,46 @@ public void onDetach() {
         super.onDetach();
         requestPermissionLauncher = null; // Clean up
     }
+
+    /**
+     *
+     */
+    private void registerResult(){
+        resultLauncher = registerForActivityResult(
+                new ActivityResultContracts.StartActivityForResult(),
+                new ActivityResultCallback<ActivityResult>() {
+                    @Override
+                    public void onActivityResult(ActivityResult result) {
+                        if (result.getData() == null) {
+                            Log.e(TAG, "No image selected.");
+                            return; // Exit early to prevent crashes
+                        }
+                        try {
+                            Uri imageUri = result.getData().getData();
+                            if (imageUri != null){
+                                uploadImage.setImageURI(imageUri);
+                                imageBase64 = imageHandler.compressImageToBase64(requireContext(), result.getData().getData());
```

Comment:
```suggestion
                                imageBase64 = ImageHandler.compressImageToBase64(requireContext(), result.getData().getData());
```
just directly use the static class